docs/hooks.md

```diff
@@ -1,15 +1,43 @@
 # Hooks
 
-When integrating tusd into an application, it is important to establish a communication channel between the two components. The tusd binary accomplishes this by providing a Hook system which will execute custom code when certain events happen, such as an upload being created or finished. While being simple, yet powerful enough, enabled uses ranging from logging over validation and authorization to processing the uploaded files.
+When integrating tusd into an application, it is important to establish a communication channel between the two components. The tusd binary accomplishes this by providing a system when triggers actions when certain events happen, such as an upload being created or finished. This simple-but-powerful system enables uses ranging from logging over validate and authorization to processing the uploaded files. 
```

Comment:
> providing a system when triggers actions when certain events happen

That's not a correct sentence. What do you wanted to say?

> ranging from logging over validate and authorization

Why do you swap the noun for a verb? IMO, the noun suited better in the list of other nouns.

Also, trailing whitespace ;)